### ISSUE TITLE
Update IPFS node to go-ipfs v0.12.0

### ISSUE THREAD

**Issue by lidel** [brave/brave-browser#21219]
go-ipfs 0.12 was released: https://github.com/ipfs/go-ipfs/releases/tag/v0.12.0

It includes no new features, only a datastore migration.

Given that Brave node has GC enabled at cache capped at 1GiB, it should not take too long – see https://github.com/brave/brave-browser/issues/20106#issuecomment-1047092536

**Comment by stephendonner** [comment]
cc @spylogsster

**Comment by stephendonner** [comment]
## Verified `PASSED` using 

Brave | 1.38.75 Chromium: 100.0.4896.60 (Official Build) beta (x86_64)
-- | --
Revision | 6a5d10861ce8de5fce22564658033b43cb7de047-refs/branch-heads/4896@{#875}
OS | macOS Version 11.6.5 (Build 20G527)

### Steps:
1. installed `1.38.75`
2. launched Brave
3. loaded `brave://ipfs`
4. clicked on `Install and restart`
5. look at the version #

#### Confirmed I got `go-ipfs` `0.12.0`

example | example
---------|---------
<img width="1607" alt="Screen Shot 2022-03-31 at 1 28 46 PM" src="https://user-images.githubusercontent.com/387249/161144960-2062c3e9-91ff-4abc-98a9-23e9de5ee82c.png"> | <img width="1493" alt="Screen Shot 2022-03-31 at 1 29 09 PM" src="https://user-images.githubusercontent.com/387249/161144842-43f33522-247e-4233-ad07-e35d4cd51644.png">

**Comment by MadhaviSeelam** [comment]
Verified `Passed` using

Brave | 1.38.75 Chromium: 100.0.4896.60 (Official Build) beta (64-bit)
-- | --
Revision | 6a5d10861ce8de5fce22564658033b43cb7de047-refs/branch-heads/4896@{#875}
OS | Windows 11 Version 21H2 (Build 22000.593)

- Installed 1.38.75
- launched Brave
- loaded brave://ipfs
- clicked on Install and restart
- look at the version #
- Confirmed I got go-ipfs 0.12.0

ex1|ex2|ex3
---|---|---
<img width="930" alt="screenshot1" src="https://user-images.githubusercontent.com/98358127/161165201-f77d1aac-9441-4283-be26-af2c51b3a532.png">|<img width="screenshot2" alt= "screenshot2" src="https://user-images.githubusercontent.com/98358127/161165272-9f88e41d-1fc0-42d3-8af1-c55336758293.png">|<img width="930" alt="screenshot3" alt="screenshot2" src="https://user-images.githubusercontent.com/98358127/161165353-797df300-fa29-47be-bd4e-a4f9d7c8aec3.png">

**Comment by stephendonner** [comment]
Verified `PASSED` using Brave `1.38.83` on Ubuntu 18.04.6 LTS, 64-bit

Steps:

1. installed `1.38.83`
2. launched Brave
3. loaded `brave://ipfs`
4. clicked on `Install and restart`
5. look at the version #

Confirmed I got `go-ipfs` `0.12.0`

example | example
---------|---------
<img width="946" alt="Screen Shot 2022-04-07 at 12 56 41 PM" src="https://user-images.githubusercontent.com/387249/162285938-fc1bada6-14ca-414e-ab95-90e99c72f19d.png"> | <img width="946" alt="Screen Shot 2022-04-07 at 12 57 01 PM" src="https://user-images.githubusercontent.com/387249/162285955-c4eda930-7b79-4627-86dc-05df5e8b1e4c.png">
